# active-motif-tipchip

This repository provides a wrapper script for demultiplexing **TIP-ChIP** sequencing data. The script accepts paired-end FASTQ files, inline barcode metadata, and configurable parameters for barcode mismatches and trimming.

---

## 📦 Installation

### 1) Install `conda` (if not already installed)
We recommend [Miniconda](https://docs.conda.io/en/latest/miniconda.html).

### 2) Install [`fqtk`](https://github.com/fulcrumgenomics/fqtk)
```bash
conda install -c bioconda fqtk
```

### 3) Clone this repository
```bash
git clone https://github.com/activemotif/active-motif-tipchip.git
cd active-motif-tipchip
chmod +x bin/run_demux_tipchip.sh
```

---

## 🚀 Usage

The wrapper script is located at `bin/run_demux_tipchip.sh`, run it with:

```bash
bash run_demux_tipchip.sh -i R1.fastq.gz R2.fastq.gz -m inlinebarcode_metadata.tsv [options]
```

### Required arguments
- `-i R1.fastq.gz R2.fastq.gz` → paired-end FASTQ files generated by a standard Illumina sequencer.
- `-m inlinebarcode_metadata.tsv` → inline barcode metadata file  

### Optional arguments
- `-o OUTPUT_DIR` → output directory (default: `FASTQ_Output`)  
- `-x MAX_MISMATCHES` → maximum mismatches (default: `1`)  
- `-d MIN_MISMATCH_DELTA` → minimum mismatch delta (default: `1`)  

### Creating the `inlinebarcode_metadata.tsv`

The `inlinebarcode_metadata.tsv` file defines the mapping between **sample IDs** and their **inline barcodes**. This file is required for the demultiplexing step. The file is a **tab-delimited text file** with the following columns:

- `sample_id` → a user-defined name for each sample (can be changed to anything meaningful, e.g. `sample01`, `patientA`, `condition1_rep1`)  
- `barcode` → the concatenated sequence of **Read 1 barcode** + **Read 2 barcode** from the TIP-ChIP plate map (`data/TIP-ChIP Plate Map with Barcodes.xlsx`)
- an example `inlinebarcode_metadata.tsv` file can be found at: `data/inlinebarcode_metadata.tsv`


---

## 📝 Example

```bash
bash run_demux_tipchip.sh \
-i sample1_r1.fastq.gz sample1_r2.fastq.gz \
-m inlinebarcode_metadata.tsv \
-o demux_results
```

This will demultiplex reads using inline barcodes and write outputs to `demux_results`.

---

## 📂 Output

Demultiplexed FASTQs are written to the specified output directory. Each sample defined in the metadata file will have its own FASTQ files.

After running this **inline barcode demultiplexing**, the resulting FASTQs are ready for downstream analysis. For example, they can be used directly as inputs to standard **ChIP-seq pipelines**.